src/api/firebase.js

```diff
@@ -187,56 +231,16 @@ export function comparePurchaseUrgency(data) {
 		sortedData.push([]);
 	}
 
-	// Loop through all items and assign urgency based on days until next purchase
-	for (let item of data) {
-		// If the item has been purchased before, return days since last purchase
-		// If the item has never been purchased, return the days since item creation
-		const daysSincePurchase = item.dateLastPurchased
-			? getDaysBetweenDates(item.dateLastPurchased)
-			: getDaysBetweenDates(item.dateCreated);
-
-		// Assign urgency
-		if (daysSincePurchase >= 60) {
-			item.purchaseStatus = 'Inactive';
-			item.urgencyCategory = 4; // Inactive
-		} else {
-			const daysUntilPurchase = item.dateLastPurchased
-				? getDaysBetweenDates(item.dateNextPurchased)
-				: getDaysBetweenDates(undefined, item.dateNextPurchased);
-
-			item.purchaseStatus = 'Active';
-			item.days = daysUntilPurchase;
-
-			if (daysUntilPurchase < 0) {
-				item.urgencyCategory = 0; // Overdue
-			} else if (daysUntilPurchase <= 7) {
-				item.urgencyCategory = 1; // Soon
-			} else if (daysUntilPurchase > 7 && daysUntilPurchase < 30) {
-				item.urgencyCategory = 2; // Kind of Soon
-			} else if (daysUntilPurchase >= 30) {
-				item.urgencyCategory = 3; // Not Soon
-			} else {
-				item.urgencyCategory = 4; // Inactive
-			}
-		}
-	}
+	//Sort items, first by urgency and then alphabetically.
+
+	const sortByUrgencyCategory = (a, b) => getItemUrgency(a) - getItemUrgency(b);
+	const sortByName = (a, b) => a.name.localeCompare(b.name);
 
-	//Sort items, first by purchase status,
-	//then by urgency, and lastly alphabetically.
-	data.sort(
-		(a, b) =>
-			a.purchaseStatus.localeCompare(b.purchaseStatus) ||
-			a.urgencyCategory - b.urgencyCategory ||
-			a.name.localeCompare(b.name),
-	);
+	data.sort(sortByName).sort(sortByUrgencyCategory);
```

Comment:
If you'd like, you can still keep the old way, since you're just abstracting away the specific logic pieces you check at every `OR`

```suggestion
	data.sort(
		(a, b) =>
			sortByUrgency(a,b) ||
			sortByName(a,b)
	);
```